# WHITE HOUSE VISIT ANALYSIS

This data was collected between a seven year gap of visit of the white house during the tenure of president Barack Obama. The aim of this analysis is to know which time of the hour had the greatest number of visits. Most used venue for visit, Year with the highest number of visits.

**Lets import the modules we will use**

In [1]:
import pandas as pd
import datetime as dt

In [2]:
wh_visit = pd.read_csv("potus_visitors_2015.csv")

In [3]:
wh_visit.head(10)

name       appt_made_date appt_start_date appt_end_date  \
0       Joshua T. Blanton  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
1         Jack T. Gutting  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
2       Bradley T. Guiles  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
3          Loryn F. Grieb  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
4        Travis D. Gordon  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
5         Taylor D. Gibbs  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
6       Dameriah A. Smith  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
7  Dylan S. Hopkinstaylor  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
8      Joseph S. Barbaria  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   
9    Jonathan L. Buckland  2014-12-18T00:00:00     1/6/15 9:30  1/6/15 23:59   

  visitee_namelast visitee_namefirst meeting_room  \
0              NaN             potus    west wing   
1              NaN             potus    west wing   
2              NaN             potus    west wing   
3              NaN             potus    west wing   
4              NaN             potus    west wing   
5              NaN             potus    west wing   
6              NaN             potus    west wing   
7              NaN             potus    west wing   
8              NaN             potus    west wing   
9              NaN             potus    west wing   

                         description  
0  JointService Military Honor Guard  
1  JointService Military Honor Guard  
2  JointService Military Honor Guard  
3  JointService Military Honor Guard  
4  JointService Military Honor Guard  
5  JointService Military Honor Guard  
6  JointService Military Honor Guard  
7  JointService Military Honor Guard  
8  JointService Military Honor Guard  
9  JointService Military Honor Guard

In [4]:
wh_visit.shape

(47954, 8)

In [5]:
wh_visit.dtypes

name                 object
appt_made_date       object
appt_start_date      object
appt_end_date        object
visitee_namelast     object
visitee_namefirst    object
meeting_room         object
description          object
dtype: object

In [6]:
wh_visit.describe()

name       appt_made_date appt_start_date  appt_end_date  \
count             47954                47954           47954          47954   
unique            37924                  353            1174            193   
top     Alan C. Prather  2015-12-14T00:00:00   6/15/15 15:15  6/15/15 23:59   
freq                 32                 1812            3064           3246   

       visitee_namelast visitee_namefirst meeting_room           description  
count              6069             47953        47954                  8053  
unique              112               102          148                   169  
top                Velz             POTUS   State Floo  military honor guard  
freq               1453             27871        22670                  1030

In [7]:
wh_visit["meeting_room"].value_counts().sort_values(ascending = False).head(20)

State Floo     22670
SOUTH LAWN      7681
South Lawn      2133
Oval Offic      2026
south lawn      1791
Residence       1536
East Room       1396
State Floor      828
STATE FLOO       553
East Wing/       521
state floo       518
SCA              513
west wing        346
Cabinet Ro       342
RESIDENCE        336
ww/Oval Of       318
Residence/       283
east room        248
Rose Garde       235
Roosevelt        235
Name: meeting_room, dtype: int64

In [8]:
wh_visit["meeting_room"] = wh_visit["meeting_room"].str.lower()
wh_visit["meeting_room"] = wh_visit["meeting_room"].str.replace("state floo", "state floor")
wh_visit["meeting_room"] = wh_visit["meeting_room"].str.replace("state floorr", "state floor")
wh_visit["meeting_room"] = wh_visit["meeting_room"].str.replace("/", "")
wh_visit["meeting_room"] = wh_visit["meeting_room"].str.replace("oval offic", "oval office")
wh_visit["meeting_room"].describe()

count           47954
unique            100
top       state floor
freq            24583
Name: meeting_room, dtype: object

In [9]:
wh_visit["meeting_room"].value_counts().sort_values(ascending = False).head(20)

state floor           24583
south lawn            11605
oval office            2265
residence              2155
east room              1646
east wing               587
west wing               585
sca                     513
rose garde              431
wwoval of               392
cabinet ro              342
oval                    281
roosevelt               254
north driv              235
230                     207
east roomeast wing      186
45                      131
state dini              122
west wing drive         116
west exec               111
Name: meeting_room, dtype: int64

*The first floor of the White House Residence is often called the "State Floor," because this is where formal receptions of state are held.*

The above showed that the State floor is the most used venue for meetings. 

The explanation of the State Floor explains why it is the most visited venue in the white house in the year 2015

*The South Lawn presents a long north-south vista from the house. Open to the public until the Second World War, it is now a closed part of the White House grounds that provides a setting for official events like the State Arrival Ceremony as well as informal gatherings including the annual White House Egg Rolling Contest and staff barbecues. Marine One, the presidential helicopter, departs from and lands on the South Lawn.*


**Next we need to determine how long the meetings take in each venue.** 

- From the data we can see that the numbers were increasing steadily till it got to 2265 and then it jumped to 11605. So it is safe to assume that most people visiting the white house do so for occasions. In analysing the amount of time the meetings take we will not be considering meetings held in the State floor and South lawn. Let's create a duplicate of our data and drop the rows where meeting room is state floor or south lawn.

In [10]:
wh_copy = wh_visit.copy(deep = True)

In [11]:
wh_copy = wh_copy[wh_copy.meeting_room != "state floor"]

In [12]:
wh_copy = wh_copy[wh_copy.meeting_room != "south lawn"]

In [36]:
number_of_visits = wh_copy["meeting_room"].value_counts().sort_values(ascending = False).head(5)
number_of_visits

oval office    2265
residence      2155
east room      1646
east wing       587
west wing       585
Name: meeting_room, dtype: int64

In [14]:
wh_copy["appt_start_date"] = pd.to_datetime(wh_copy["appt_start_date"],format="%m/%d/%y %H:%M")
# wh_copy["appt_start_date"] = wh_copy["appt_start_date"].dt.strftime("%d/%m/%Y %H")
# wh_copy["appt_start_date"] = wh_copy["appt_start_date"].astype("datetime64[ns]")

In [15]:
wh_copy["appt_end_date"] = pd.to_datetime(wh_copy["appt_end_date"],format="%m/%d/%y %H:%M")
# wh_copy["appt_end_date"] = wh_copy["appt_end_date"].dt.strftime("%d/%m/%Y %H")
# wh_copy["appt_end_date"] = wh_copy["appt_end_date"].astype("datetime64[ns]")

In [16]:
wh_copy.head(10)

name       appt_made_date     appt_start_date  \
0       Joshua T. Blanton  2014-12-18T00:00:00 2015-01-06 09:30:00   
1         Jack T. Gutting  2014-12-18T00:00:00 2015-01-06 09:30:00   
2       Bradley T. Guiles  2014-12-18T00:00:00 2015-01-06 09:30:00   
3          Loryn F. Grieb  2014-12-18T00:00:00 2015-01-06 09:30:00   
4        Travis D. Gordon  2014-12-18T00:00:00 2015-01-06 09:30:00   
5         Taylor D. Gibbs  2014-12-18T00:00:00 2015-01-06 09:30:00   
6       Dameriah A. Smith  2014-12-18T00:00:00 2015-01-06 09:30:00   
7  Dylan S. Hopkinstaylor  2014-12-18T00:00:00 2015-01-06 09:30:00   
8      Joseph S. Barbaria  2014-12-18T00:00:00 2015-01-06 09:30:00   
9    Jonathan L. Buckland  2014-12-18T00:00:00 2015-01-06 09:30:00   

        appt_end_date visitee_namelast visitee_namefirst meeting_room  \
0 2015-01-06 23:59:00              NaN             potus    west wing   
1 2015-01-06 23:59:00              NaN             potus    west wing   
2 2015-01-06 23:59:00              NaN             potus    west wing   
3 2015-01-06 23:59:00              NaN             potus    west wing   
4 2015-01-06 23:59:00              NaN             potus    west wing   
5 2015-01-06 23:59:00              NaN             potus    west wing   
6 2015-01-06 23:59:00              NaN             potus    west wing   
7 2015-01-06 23:59:00              NaN             potus    west wing   
8 2015-01-06 23:59:00              NaN             potus    west wing   
9 2015-01-06 23:59:00              NaN             potus    west wing   

                         description  
0  JointService Military Honor Guard  
1  JointService Military Honor Guard  
2  JointService Military Honor Guard  
3  JointService Military Honor Guard  
4  JointService Military Honor Guard  
5  JointService Military Honor Guard  
6  JointService Military Honor Guard  
7  JointService Military Honor Guard  
8  JointService Military Honor Guard  
9  JointService Military Honor Guard

In [17]:
wh_copy["t_diff"] = wh_copy["appt_end_date"] - wh_copy["appt_start_date"]

In [18]:
wh_copy.head(10)

name       appt_made_date     appt_start_date  \
0       Joshua T. Blanton  2014-12-18T00:00:00 2015-01-06 09:30:00   
1         Jack T. Gutting  2014-12-18T00:00:00 2015-01-06 09:30:00   
2       Bradley T. Guiles  2014-12-18T00:00:00 2015-01-06 09:30:00   
3          Loryn F. Grieb  2014-12-18T00:00:00 2015-01-06 09:30:00   
4        Travis D. Gordon  2014-12-18T00:00:00 2015-01-06 09:30:00   
5         Taylor D. Gibbs  2014-12-18T00:00:00 2015-01-06 09:30:00   
6       Dameriah A. Smith  2014-12-18T00:00:00 2015-01-06 09:30:00   
7  Dylan S. Hopkinstaylor  2014-12-18T00:00:00 2015-01-06 09:30:00   
8      Joseph S. Barbaria  2014-12-18T00:00:00 2015-01-06 09:30:00   
9    Jonathan L. Buckland  2014-12-18T00:00:00 2015-01-06 09:30:00   

        appt_end_date visitee_namelast visitee_namefirst meeting_room  \
0 2015-01-06 23:59:00              NaN             potus    west wing   
1 2015-01-06 23:59:00              NaN             potus    west wing   
2 2015-01-06 23:59:00              NaN             potus    west wing   
3 2015-01-06 23:59:00              NaN             potus    west wing   
4 2015-01-06 23:59:00              NaN             potus    west wing   
5 2015-01-06 23:59:00              NaN             potus    west wing   
6 2015-01-06 23:59:00              NaN             potus    west wing   
7 2015-01-06 23:59:00              NaN             potus    west wing   
8 2015-01-06 23:59:00              NaN             potus    west wing   
9 2015-01-06 23:59:00              NaN             potus    west wing   

                         description   t_diff  
0  JointService Military Honor Guard 14:29:00  
1  JointService Military Honor Guard 14:29:00  
2  JointService Military Honor Guard 14:29:00  
3  JointService Military Honor Guard 14:29:00  
4  JointService Military Honor Guard 14:29:00  
5  JointService Military Honor Guard 14:29:00  
6  JointService Military Honor Guard 14:29:00  
7  JointService Military Honor Guard 14:29:00  
8  JointService Military Honor Guard 14:29:00  
9  JointService Military Honor Guard 14:29:00

In [19]:
meeting_room = wh_copy["meeting_room"].value_counts()
first_5_room = meeting_room[meeting_room > 550].index
first_5_room

Index(['oval office', 'residence', 'east room', 'east wing', 'west wing'], dtype='object')

In [31]:
meeting_room_sum = {}
meeting_room_mean = {}

for room in first_5_room:
    room_only = wh_copy[wh_copy["meeting_room"] == room]
    room_only_sum = room_only["t_diff"].sum()
    room_only_mean = room_only["t_diff"].mean()
    meeting_room_sum[room] = room_only_sum
    meeting_room_mean[room] = room_only_mean
    
meeting_room_sum
meeting_room_mean

{'oval office': Timedelta('0 days 10:14:44.503311'),
 'residence': Timedelta('0 days 07:23:49.364269'),
 'east room': Timedelta('0 days 11:45:29.307411'),
 'east wing': Timedelta('0 days 15:46:56.320272'),
 'west wing': Timedelta('0 days 13:14:21.025641')}

In [32]:
meeting_hours = pd.Series(meeting_room_sum)
meeting_hours_avg = pd.Series(meeting_room_mean)


In [37]:
meeting_hours_per_room = pd.DataFrame(meeting_hours, columns = ["time_spent_per_room"])
meeting_hours_per_room["avg_time_spent"] = meeting_hours_avg
meeting_hours_per_room["Number_of_visits"] = number_of_visits
meeting_hours_per_room

time_spent_per_room  avg_time_spent  Number_of_visits
oval office   966 days 22:30:00 10:14:44.503311              2265
residence     664 days 04:38:00 07:23:49.364269              2155
east room     806 days 09:54:00 11:45:29.307411              1646
east wing     386 days 00:13:00 15:46:56.320272               587
west wing     322 days 16:55:00 13:14:21.025641               585

From the above the data reads from the 6th of January 2015 to 18th of December 2015. The Oval Office was visited the most also is the room that has the highest total sum of hours spent. However, the average number of hours spent in a meeting in the Oval Office is less than the East wing and East room. 

- *The East Wing is a part of the White House Complex. It is a two-story structure east of the White House Executive Residence, the home of the president of the United States. While the West Wing generally serves the president's executive office staff, the East Wing serves as office space for the first lady and her staff, including the White House social secretary, White House Graphics and Calligraphy Office and correspondence staff. The East Wing also includes the visitors' entrance, and the East Colonnade, a corridor connecting the body of the East Wing to the residence. Along the corridor is the White House theater, also called the Family theater. Social and touring visitors to the White House usually enter through the East Wing.*


- *The East Room is an event and reception room in the White House, the home of the President of the United States. The East Room is the largest room in the Executive Mansion; it is used for dances, receptions, press conferences, ceremonies, concerts, and banquets.*


**Since the East Room is the Residential arm of the White House, this explains why less people visit the but spend more time since it is the room with highest mean hours.**

**The East wing is used for receptions and ceremonies, this gatherings usually take longer than short meetings but don't occur often hence it has an high average time with less visit.**


In [39]:
wh_visit["appt_start_date"] = pd.to_datetime(wh_visit["appt_start_date"],format="%m/%d/%y %H:%M")

In [43]:
wh_visit.head()

name       appt_made_date     appt_start_date appt_end_date  \
0  Joshua T. Blanton  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
1    Jack T. Gutting  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
2  Bradley T. Guiles  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
3     Loryn F. Grieb  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
4   Travis D. Gordon  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   

  visitee_namelast visitee_namefirst meeting_room  \
0              NaN             potus    west wing   
1              NaN             potus    west wing   
2              NaN             potus    west wing   
3              NaN             potus    west wing   
4              NaN             potus    west wing   

                         description  
0  JointService Military Honor Guard  
1  JointService Military Honor Guard  
2  JointService Military Honor Guard  
3  JointService Military Honor Guard  
4  JointService Military Honor Guard

In [46]:
wh_visit["appt_month"] = wh_visit["appt_start_date"].dt.strftime("%B")

In [47]:
wh_visit.head()

name       appt_made_date     appt_start_date appt_end_date  \
0  Joshua T. Blanton  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
1    Jack T. Gutting  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
2  Bradley T. Guiles  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
3     Loryn F. Grieb  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   
4   Travis D. Gordon  2014-12-18T00:00:00 2015-01-06 09:30:00  1/6/15 23:59   

  visitee_namelast visitee_namefirst meeting_room  \
0              NaN             potus    west wing   
1              NaN             potus    west wing   
2              NaN             potus    west wing   
3              NaN             potus    west wing   
4              NaN             potus    west wing   

                         description appt_start_date_month appt_month  
0  JointService Military Honor Guard               January    January  
1  JointService Military Honor Guard               January    January  
2  JointService Military Honor Guard               January    January  
3  JointService Military Honor Guard               January    January  
4  JointService Military Honor Guard               January    January

In [49]:
wh_visit["appt_month"].value_counts(dropna = False)

December     13029
June          7743
April         4996
September     4416
October       3669
May           3013
July          2930
March         2262
February      2165
August        1350
January       1248
November      1133
Name: appt_month, dtype: int64

The Month with the highest number of visit is December in the year 2015. 

# Conclusions

**The first floor of the White House Residence is often called the "State Floor," because this is where formal receptions of state are held**


**Excluding the State floor and the South lawn, the Oval Office is the mostly visited room in the White House. Although, the East room is where people spent the highest amount of visiting hours since it is the residential arm of the White House.**


**The Month of December has the highest number of visit in the year 2015.**